examples/abitest/oak_app_manifest.toml

```diff
@@ -1,9 +1,9 @@
 name = "abitest"
 
 [modules]
-frontend_module = { path = "examples/abitest/bin/abitest_0_frontend.wasm" }
-backend_module = { path = "examples/abitest/bin/abitest_1_backend.wasm" }
-linear_handles_module = { path = "examples/abitest/bin/abitest_linear_handles.wasm" }
+frontend_module = { path = "examples/target/wasm32-unknown-unknown/release/abitest_0_frontend.wasm" }
+backend_module = { path = "examples/target/wasm32-unknown-unknown/release/abitest_1_backend.wasm" }
+linear_handles_module = { path = "examples/target/wasm32-unknown-unknown/release/abitest_linear_handles.wasm" }
```

Comment:
This doesn't look nice at all. I clearly prefer the previous style and I am unsure why this change is needed here.

Reply:
This is the canonical location where the artifacts are built. The previous version was compiling everything twice , one for tests and one for the examples, without even reusing the intermediate cache. Basically test code and runner were doing the same job twice. The previous version was also not reusing cache between examples, because it ignored the workspace structure in which the examples are defined. Let me clarify in the commit message and in code comments.

Reply:
How does the output path of the binary influences that? This feels strange

Reply:
It's the target dir that affects it, maybe we can get the best of both by making the bin dir relative to the workspace root, let me try that

Reply:
PTAL now, I think now it is a compromise between short paths and actually sharing caches in a meaningful way (though personally I still prefer the longer more explicit paths -- but let's have a separate discussion about that perhaps).